javascript/spec/game.spec.js

```diff
@@ -13,22 +13,10 @@ describe("The test environment", function() {
 });
 
 describe("The game", function() {
-  let log = [];
-
-  beforeAll(function () {
-    console.log = function (message) {
-      log.push(message);
-    }
-  })
-
-  afterEach(function () {
-    log = [];
-  })
-
   using("seeds 1 to 1000", new Array(1000).fill('').map((_, index) => [index + 1]), function (seed) {
     it(`should produce the expected output when seeded with ${seed}`, function () {
-      runner.play(seed);
-      expect(log.join("\n")).toEqual(fs.readFileSync(`./spec/fixtures/output-seeded-with-${seed}.txt`).toString());
+      const game = runner.play(seed);
+      expect(game.logs.join("\n")).toEqual(fs.readFileSync(`./spec/fixtures/output-seeded-with-${seed}.txt`).toString());
```

Comment:
I really hated monkey-patching `console.log`, since that could potentially have unwanted consequences in the future (ie async nature of console.log), so I changed how the game from using console.log to this.log, which keeps track of it's own logs, as well as using `console.log`. Seemed cleaner to me, but it did required changing the tests slightly.